### PR TITLE
[NPU] add causal conv1d for ascend kda backend

### PR DESCRIPTION
**Pull request by zhaozx-cn**
## Motivation
the **causal_conv1d_linear_verify_npu + speculative_state_scatter_npu** mechanism performs poorly on kda spec branch since it requires using intermediate_conv_window, which introduces more read/write operations. we propose replacing it with  **causal_conv1d + conv_state_rollback** mechanism that operates in-place on the conv state buffers. in addition, causal_conv1d ascendc op can also replace the current torch native or triton implementations used in prefill/decode branches to boost performance. this approach also aligns with the conv stage of the ascend gdn backend.

## Modifications

1. replace causal_conv1d_linear_verify_npu + speculative_state_scatter_npu with causal_conv1d + conv_state_rollback for kda spec branch.

2. use causal_conv1d for kda prefill and decode branch.
3. resolve host bound in update_mamba_state_after_mtp_verify.

## Accuracy Tests

gpqa 
official : 93.5
this pr : 93.9

## Speed Tests and Profiling

**configuration:** 8k in, 1k out, ep=64, DSPARK=7

| Metric               | Before   | After    | Improvement |
|----------------------|----------|----------|-------------|
| TTFT (ms)            | 7364.2   | 6147.8   | -16.5%      |
| TPOT (ms)            | 15.05    | 12.68    | -15.7%      |

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
4. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
5. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #33069406580](https://github.com/sgl-project/sglang/actions/runs/33069406580)
Latest PR Test (Extra): :x: [Run #33069406342](https://github.com/sgl-project/sglang/actions/runs/33069406342)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33069406514](https://github.com/sgl-project/sglang/actions/runs/33069406514)

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py [hunk 1]
```diff
@@ -154,13 +181,16 @@ def forward_decode(
...
         cache_indices = self.forward_metadata.mamba_cache_indices
 
-        qkv = causal_conv1d_update_npu(
-            mixed_qkv,
-            conv_states,
-            layer.conv_weights,
-            layer.bias,
-            activation="silu",
-            conv_state_indices=cache_indices,
+        qkv = torch.ops.npu.causal_conv1d(
```
Comment (iforgetmyname) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py]: add comment: activation silu only

File: python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py [hunk 2]
```diff
@@ -154,13 +181,16 @@ def forward_decode(
...
+            mixed_qkv.contiguous(),
+            self._get_conv_weights_t(layer, mixed_qkv.dtype),
+            conv_states=conv_states,
+            bias=layer.bias,
+            query_start_loc=query_start_loc,
```
Comment (zhaozx-cn) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py:190]: add note about activation.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py [hunk 3]
```diff
@@ -245,37 +275,27 @@ def forward_extend(
...
-            q, q_conv_weight, q_bias, q_conv_state, **conv_kwargs
-        )
-        k = self._causal_conv1d_extend(
-            k, k_conv_weight, k_bias, k_conv_state, **conv_kwargs
-        )
-        v = self._causal_conv1d_extend(
-            v, v_conv_weight, v_bias, v_conv_state, **conv_kwargs
+            cache_indices=cache_indices,
+            has_initial_state=has_initial_state,
+            activation_mode=1,
```
Comment (zhaozx-cn) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py:294]: add note about activation.

File: python/sglang/srt/hardware_backend/npu/memory_pool_npu.py [hunk 4]
```diff
@@ -32,18 +32,19 @@ def _init_npu_conv_state(
...
+    # speculative_num_draft_tokens - 1 so that verify can write all draft
+    # token conv states directly into conv_states (GDN rollback scheme).
     conv_state = [
         torch.zeros(
             size=(
                 conv_state_in.shape[0],
                 conv_state_in.shape[1],
-                conv_shape[1] if is_kda else conv_shape[1] + extra_conv_len,
-                conv_shape[0],
+                (conv_shape[0] if is_kda else conv_shape[1]) + extra_conv_len,
```
Comment (zhaozx-cn) [inline python/sglang/srt/hardware_backend/npu/memory_pool_npu.py:46]: maybe need reshape here.

Comment (zhaozx-cn) [reply python/sglang/srt/hardware_backend/npu/memory_pool_npu.py:46]: both gdn and kda arrive as (window, channel) for causal conv1d

File: python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py (hunk 2 above)
Comment (zhaozx-cn) [reply python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py:190]: done

File: python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py (hunk 3 above)
Comment (zhaozx-cn) [reply python/sglang/srt/hardware_backend/npu/attention/ascend_kda_backend.py:294]: done
